crates/swc_ecma_transforms_module/src/common_js.rs

```diff
@@ -55,6 +58,57 @@ where
     }
 }
 
+struct LazyIdentifierVisitor {
+    scope: Rc<RefCell<Scope>>,
+    top_level_idents: AHashSet<JsWord>,
+}
+impl LazyIdentifierVisitor {
+    fn new(scope: Rc<RefCell<Scope>>) -> Self {
+        LazyIdentifierVisitor {
+            scope,
+            top_level_idents: Default::default(),
+        }
+    }
+}
+
+impl Visit for LazyIdentifierVisitor {
+    noop_visit_type!();
+
+    fn visit_import_decl(&mut self, _: &ImportDecl) {}
+    fn visit_export_decl(&mut self, _: &ExportDecl) {}
+    fn visit_named_export(&mut self, _: &NamedExport) {}
+    fn visit_export_default_decl(&mut self, _: &ExportDefaultDecl) {}
+    fn visit_export_default_expr(&mut self, _: &ExportDefaultExpr) {}
+
+    fn visit_export_all(&mut self, export: &ExportAll) {
+        self.top_level_idents.insert(export.src.value.clone());
+    }
+
+    fn visit_function(&mut self, _: &Function) {}
+    fn visit_constructor(&mut self, _: &Constructor) {}
+    fn visit_setter_prop(&mut self, _: &SetterProp) {}
+    fn visit_getter_prop(&mut self, _: &GetterProp) {}
+    fn visit_class_prop(&mut self, _: &ClassProp) {}
+
```

Comment:
I think having a visitor method for `PropName` is preferrable.

Reply:
It visits children iff it's `PropName::Computed`.

Reply:
I don't understand what you mean here, can you elaborate please?

Reply:
I think you're saying that we need to visit the child of an expression if it's like `obj[prop]`, yes?

Reply:
I meant it would be good to have a method `visit_prop_name`, while it does

```rust

match n {
    PropName::Computed(n) => {
        n.visit_with(self);
    }
    _ => {}
}

```

This is to prevent accidental breakage due to AST changes.